src/KubernetesClient/Authentication/GcpTokenProvider.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Diagnostics;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using k8s.Exceptions;
+using Microsoft.Rest;
+using Newtonsoft.Json.Linq;
+
+namespace k8s.Authentication
+{
+    public class GcpTokenProvider : ITokenProvider
```

Comment:
I assume this will get removed when your other PR merges.

Reply:
The PR is rebased on top of my other PR so it shows up here. I'll rebase again once the other PR goes in (which I assume will be the first one to be accepted given its smaller scope)